chaoslib/activity.py

```diff
@@ -143,10 +143,12 @@ def execute_activity(experiment: Experiment, activity: Activity,
                   configuration=configuration, secrets=secrets) as control:
         pauses = activity.get("pauses", {})
         pause_before = pauses.get("before")
-        if pause_before:
-            logger.info("Pausing before next activity for {d}s...".format(
-                d=pause_before))
-            time.sleep(pause_before)
+        # only pause when not in dry-mode
+        if not dry:
+            if pause_before:
+                logger.info("Pausing before next activity for {d}s...".format(
+                    d=pause_before))
+                time.sleep(pause_before)
```

Comment:
would it be better to combine the two conditions to reduce complexity? or even add the 'and dry' to the setting of pause_before?